blocks/article-meta/article-meta.js

```diff
@@ -0,0 +1,177 @@
+import { getLibs } from '../../scripts/utils.js';
+
+async function validateAuthorUrl(url) {
+  if (!url) return null;
+
+  const resp = await fetch(`${url.toLowerCase()}.plain.html`);
+  if (!resp?.ok) {
+    console.log(`Could not retrieve metadata for ${url}`);
+    return null;
+  }
+
+  const html = await resp.text();
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(html, 'text/html');
+
+  return doc;
+}
+
+function openPopup(e) {
+  const target = e.target.closest('a');
+  const href = target.getAttribute('data-href');
+  const type = target.getAttribute('data-type');
+  window.open(
+    href,
+    type,
+    'popup,top=233,left=233,width=700,height=467',
+  );
+}
+
+async function buildAuthorInfo(authorEl, bylineContainer) {
+  const miloLibs = getLibs();
+  const { createTag, getConfig } = await import(`${miloLibs}/utils/utils.js`);
+
+  const { href, textContent } = authorEl;
+
+  const config = getConfig();
+  const base = config.miloLibs || config.codeRoot;
+  const authorImg = createTag('div', { class: 'article-author-image' });
+  authorImg.style.backgroundImage = `url(${base}/blocks/article-header/adobe-logo.svg)`;
+  bylineContainer.prepend(authorImg);
+
+  const doc = await validateAuthorUrl(href);
+  if (!doc) {
+    const p = createTag('p', null, textContent);
+    authorEl.replaceWith(p);
+    return;
+  }
+
+  const img = doc.querySelector('img');
+  if (img) {
+    img.setAttribute('alt', authorEl.textContent);
+    authorImg.append(img);
+    if (!img.complete) {
+      img.addEventListener('load', () => {
+        authorImg.style.backgroundImage = 'none';
+      });
+      img.addEventListener('error', () => {
+        img.remove();
+      });
+    } else {
+      authorImg.style.backgroundImage = 'none';
+    }
+  }
+}
+
+export async function initSharingLinkFunction(sharing) {
+  const miloLibs = getLibs();
+  const { getConfig } = await import(`${miloLibs}/utils/utils.js`);
+  const { replaceKey } = await import(`${miloLibs}/features/placeholders.js`);
+  const { copyToClipboard } = await import(`${miloLibs}/utils/tools.js`);
```

Comment:
parallelize these

Reply:
reused social block from Milo as suggested by Ryan so longer relevant, closing this